src/main/java/be/appwise/core/networking/proxyman/Message.kt

```diff
@@ -0,0 +1,49 @@
+package be.appwise.core.networking.proxyman
+
+import android.os.Build
+import android.provider.Settings
+
+
+/**
+ * if requestbody is filled in succes ?
+ * */
+class Message(
+    val id: String,
+    val messageType: String,
+    val buildVersion: String = "0.9.0",
+    var content: Data?
+) {
+
+    class Project {
+        companion object {
+            val current = Project()
+        }
+
+        val name: String = ProxyManNetworkDiscoveryManager.getAppContext().applicationInfo.loadLabel(ProxyManNetworkDiscoveryManager.getAppContext().packageManager).toString()
```

Comment:
Be careful by using default values like this for an object. Especially when you wish to use the Context in a Library. Providing the `Context` like this is can create an unintentional leak which may even crash the Application.

I'd prefer you try to do the same as we did with the `BagelMessage`, just have a `data class` of type `Device` where you add the values to it. I understand a couple of things may need to be refactored to let it work. But it would be easier and cleaner IMHO.

Reply:
I followed the way Atlantis did it! Proxyman isn't as lenient with faults in the format of messages. I would like to let it be closed off from changes. You can't start the proxymanservice without setting the app context . See https://github.com/appwise-labs/AndroidCore/blob/fe2d6b9a8a8220486762c4567702f1d37b9ddf02/src/main/java/be/appwise/core/networking/proxyman/ProxyManNetworkDiscoveryManager.kt#L74-L90

Reply:
Hmm, I still have a feeling this can be improved upon, but I'm just not aware of the limitations of Proxyman/Atlantis so I'll leave it to your expertise 😉